### ISSUE TITLE
SMTP login info url issue

### ISSUE THREAD

**Issue by ninjamonkey198206** [containrrr/shoutrrr#400]
I saw others having issues with Watchtower, which I surprisingly did not, but I had a hell of a time setting the URL for SMTP notifications for Scrutiny, which uses Shoutrrr.

It turns out that the issue was that my username and password (in the URL) contained special chars (user@blahblah.com)(Wtf!snTthi$W0rking).

Just a heads up that using the HTML encoding equivalents resolved the issue, but I didn't find any reference to that possibility anywhere. Maybe I just missed it. 

Thankfully a very helpful person suggested trying the swap and that worked.

**Comment by ninjamonkey198206** [comment]
Derp. I figured out that the password used one of the delimiters that you use in the URL, which caused the issue.

Maybe a note in the docs about making sure passwords don't include those and something similar about usernames. To save someone else from the deep moment I had.

**Comment by piksel** [comment]
Yeah, I planned to add a URL builder helper to the docs that does the required escaping, but it got blocked for various reasons.

Perhaps it should also be explicitly stated that all URLs need to be properly escaped to be parsed correctly...
